redisinsight/ui/src/pages/settings/components/general-settings/datetime-formatter/DateTimeFormatter.tsx

```diff
@@ -0,0 +1,270 @@
+    const { valid, error: errorMsg } = checkDateTimeFormat(format, timezone)
+    if (!valid) {
+      setError(errorMsg || 'This format is not supported')
+      setPreview('Invalid Format')
+    } else {
+      setError('')
+      const newPreview = formatTimestamp(new Date(), format, timezone)
+      setPreview(newPreview)
+    }
+    return valid
+  }
+
+  const onRadioOptionChange = (id: string) => {
+    formik.setFieldValue('selectedRadioOption', id)
+    if (id === DatetimeRadioOption.Custom) {
+      formik.setFieldValue('customFormat', formik.values.format)
+    } else {
+      formik.setFieldValue('format', formik.values.commonFormat)
+      formik.handleSubmit()
+    }
+  }
+
+  const onTimezoneChange = (value: string) => {
+    formik.setFieldValue('timezone', value)
+
+    sendEventTelemetry({
+      event: TelemetryEvent.SETTINGS_TIME_ZONE_CHANGED,
+      eventData: {
+        currentTimezone: value,
+      }
+    })
+    formik.handleSubmit()
+  }
+
+  const onCustomFormatChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target
+    formik.setFieldValue('customFormat', value)
+    formik.setFieldValue('format', value)
+    handleFormatCheck(value)
+  }
+
+  const onCommonFormatChange = (value: string) => {
+    formik.setFieldValue('commonFormat', value)
+    formik.setFieldValue('format', value)
+
+    sendEventTelemetry({
+      event: TelemetryEvent.SETTINGS_DATE_TIME_FORMAT_CHANGED,
+      eventData: {
+        currentFormat: value,
+      }
+    })
+    formik.handleSubmit()
+  }
+
+  const onCustomFormatSubmit = () => {
+    sendEventTelemetry({
+      event: TelemetryEvent.SETTINGS_DATE_TIME_FORMAT_CHANGED,
+      eventData: {
+        currentFormat: formik.values.customFormat,
+      }
+    })
+
+    formik.handleSubmit()
+  }
+
+  const dateTimeFormatOptions: EuiRadioGroupOption[] = [
+    {
+      id: DatetimeRadioOption.Common,
+      label: (
+        <div className={styles.radioLabelWrapper}>
+          <div className={styles.radioLabelTextContainer}>
+            <EuiText color="subdued" className={styles.radioLabelText}>Pre-selected formats</EuiText>
+          </div>
+          <EuiSuperSelect
+            className={styles.datetimeInput}
+            options={dateTimeOptions}
+            valueOfSelected={formik.values.commonFormat}
+            onChange={(option) => onCommonFormatChange(option)}
+            disabled={formik.values.selectedRadioOption !== DatetimeRadioOption.Common}
+            data-test-subj="select-datetime"
+          />
+        </div>
+      ),
+    },
+    {
+      id: DatetimeRadioOption.Custom,
+      label: (
+        <div className={styles.radioLabelWrapper}>
+          <div className={styles.radioLabelTextContainer}>
+            <EuiText color="subdued" className={styles.radioLabelText}>Custom</EuiText>
+          </div>
+          {formik.values.selectedRadioOption === DatetimeRadioOption.Custom ? (
```

Comment:
you can use just `{DatetimeRadioOption.Custom && (...)}`

Reply:
done